app/src/main/java/com/github/se/wanderpals/ui/navigation/NavigationActions.kt

```diff
@@ -178,17 +178,31 @@ data class NavigationActions(
     variables.currentSuggestion = suggestion
   }
 
+  /**
+   * Serializes the navigation variables into a string format.
+   *
+   * @return The serialized string containing navigation variables.
+   */
   fun serializeNavigationVariable(): String {
-    return "currentTrip: ${variables.currentTrip}, " +
-        "latitude: ${variables.currentGeoCords.latitude}, " +
-        "longitude: ${variables.currentGeoCords.longitude}, " +
-        "currentAddress: ${variables.currentAddress}, " +
-        "suggestionId: ${variables.suggestionId}"
+    val navActionsVariablesToString =
+        arrayOf(
+            "currentTrip: ${variables.currentTrip}",
+            "latitude: ${variables.currentGeoCords.latitude}",
+            "longitude: ${variables.currentGeoCords.longitude}",
+            "currentAddress: ${variables.currentAddress}",
+            "suggestionId: ${variables.suggestionId}")
+
+    return navActionsVariablesToString.joinToString("|")
```

Comment:
A bit overkill for my liking but It's a matter of preference, string concatenation is completely fine here